llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -504,7 +504,8 @@ void VPWidenCallRecipe::execute(VPTransformState &State) {
   for (unsigned Part = 0; Part < State.UF; ++Part) {
     SmallVector<Type *, 2> TysForDecl;
     // Add return type if intrinsic is overloaded on it.
-    if (isVectorIntrinsicWithOverloadTypeAtArg(VectorIntrinsicID, -1)) {
+    if (isVectorIntrinsicWithOverloadTypeAtArg(VectorIntrinsicID, -1) &&
```

Comment:
this is a LV bug, so I think it deserves a separate commit with target independent test, probably extension to the tests added in https://github.com/llvm/llvm-project/pull/73941 as all of the tests there were using functions  returning a value.

Reply:
This fix has been moved to a separate PR as suggested: #76274